lib/nostrum/struct/emoji.ex

```diff
@@ -1,50 +1,166 @@
 defmodule Nostrum.Struct.Emoji do
-  @moduledoc """
+  @moduledoc ~S"""
   Struct representing a Discord emoji.
+
+  ## Using Emojis in Messages
+
+  A `Nostrum.Struct.Emoji` can be used in message content using the `String.Chars` 
+  protocol or `format_mention/1`. We highly recommend using the `String.Chars` protocol 
+  instead of the latter.
+
+  ```Elixir
+  emoji = %Nostrum.Struct.Emoji{id: 43819043108, name: "foxbot"}
+
+  Nostrum.Api.create_message!(189098431762321, "#{emoji}")
+  Nostrum.Api.create_message!(189098431762321, "#{Nostrum.Struct.Emoji.format_mention(emoji)}")
+  ```
+
+  ## Using Emojis in the Api
+
+  See `t:Nostrum.Struct.Emoji.emoji_api_name/0`.
+  """
+
+  alias Nostrum.Struct.Snowflake
+  alias Nostrum.Struct.User
+  alias Nostrum.Util
+
+  defstruct [
+    :id,
+    :name,
+    :user,
+    :require_colons,
+    :managed,
+    :animated,
+    :roles
+  ]
+
+  defimpl String.Chars do
+    def to_string(emoji), do: @for.format_mention(emoji)
+  end
+
+  @typedoc ~S"""
+  Emoji string to be used with the Discord API.
+
+  Some API endpoints take an `emoji`. If it is a custom emoji, it must be 
+  structured as `"id:name"`. If it is an unicode emoji, it can be structured 
+  as any of the following: 
+
+    * `"name"`
+    * A base 16 unicode emoji string.
+    * A URI encoded string.
+
+  We suggest library users to use the `get_api_name/1` function to get a 
+  `Nostrum.Struct.Emoji`'s emoji api name.
+
+  ## Examples
+
+  ```Elixir
+  # Custom Emojis
+  "nostrum:431890438091489"
+
+  # Unicode Emojis
+  "≡ƒæì"
+  "\xF0\x9F\x98\x81"
+  URI.encode("\u2b50")
```

Comment:
We don't need the `URI.encode` here anymore, but I think it's a good example of using codepoints w/ the sigil so I think the rest of the example can stay here.

Reply:
Removed in latest commit.